jerry-main/main-unix.c

```diff
@@ -419,13 +423,27 @@ context_alloc (size_t size,
 static void
 init_engine (jerry_init_flag_t flags, /**< initialized flags for the engine */
              bool debug_server, /**< enable the debugger init or not */
+             const char *debug_server_type, /**< choose which type or server create */
```

Comment:
This argument shouldn't be a string. As all "message transmission interfaces" have the same prototype, I think this function should have a function pointer argument, something like: `bool (*jerryx_debugger_transport_create) (uint16_t port)`. And the call site should pass either `jerryx_debugger_tcp_create` or `jerryx_debugger_bluetooth_create`.

Reply:
I'd also prefer the function pointer argument.